packages/react-liveness/src/components/FaceLivenessDetector/FaceLivenessDetectorCore.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { useActor, useInterpret } from '@xstate/react';
+import {
+  livenessMachine,
+  FaceLivenessDetectorCoreProps as FaceLivenessDetectorPropsFromUi,
+} from './service';
+import { View, Flex } from '@aws-amplify/ui-react';
+
+import { FaceLivenessDetectorProvider } from './providers';
+import { StartLiveness } from './StartLiveness';
+import { LivenessCheck } from './LivenessCheck';
+import { getVideoConstraints } from './StartLiveness/helpers';
+import { StartScreenComponents } from './shared/DefaultStartScreenComponents';
+import { LivenessDisplayText } from './displayText';
+import { getDisplayText } from './utils/getDisplayText';
+import { CheckScreenComponents } from './shared/FaceLivenessErrorModal';
+
+const DETECTOR_CLASS_NAME = 'liveness-detector';
+
+export type FaceLivenessDetectorComponents = StartScreenComponents &
+  CheckScreenComponents;
+
+export interface FaceLivenessDetectorCoreProps
+  extends FaceLivenessDetectorPropsFromUi {
+  components?: FaceLivenessDetectorComponents;
+  displayText?: LivenessDisplayText;
+}
+
+export default function FaceLivenessDetectorCore(
```

Comment:
Just to clarify, with this we will not need to have a dependency on `aws-amplify` when using `FaceLivenessDetectorCore` anymore right?

Reply:
To be clear, the `aws-amplify` dependency will be installed in `node_modules` but should be treeshaken out. @thaddmt Can you confirm by running this through Webpack bundle analyzer?

Reply:
What does that mean? Would consumers need to install `aws-amplify` along with amplify-ui? Or does the amplify-ui liveness have a direct dependency on `aws-amplify` (clarification on how will `aws-amplify` be installed in node_modules automatically)

Reply:
Verified it is tree shaken out, screenshot attached 

<img width="2048" alt="Screenshot 2023-05-31 at 3 38 21 PM" src="https://github.com/aws-amplify/amplify-ui/assets/68032955/09cdcfff-12ac-4da8-afb0-728978526e7a">

Reply:
To your quesiton @ayush987goyal that's correct, we already ask customers to manually install `aws-amplify` so if they were to use `credentialProvider` feature then they would not need to install that package